lib.coffee

```diff
@@ -308,8 +308,18 @@ class globals.Document
 
       fieldsWalker @meta.fields
 
-      referencesIncludeProjection = LocalCollection._compileProjection referencesInclude
       referencesExcludeProjection = LocalCollection._compileProjection referencesExclude
+      referencesIncludeModification = (document) ->
+        # We can't use _compileProjection here as this would cause top-level fields to
+        # be overwritten when updating subfields.
+        result = {}
+        for path, value of referencesInclude
+          continue if value isnt 1
+
+          value = _.reduce path.split('.'), ((doc, atom) -> doc[atom]), document
+          result[path] = value unless _.isUndefined value
```

Comment:
Where you have this? In tests? I see that we do not have tests for reference fields inside arrays currently?